core/src/com/unciv/models/Spy.kt

```diff
@@ -112,15 +112,17 @@ class Spy private constructor() : IsPartOfGameInfoSerialization {
                 if (!getCity().civ.isMajorCiv()) return
 
                 val stealableTechs = espionageManager.getTechsToSteal(getCity().civ)
-                if (stealableTechs.isEmpty()) return
+                if (stealableTechs.isEmpty() || getTurnsRemainingToStealTech() < 0) return
                 setAction(SpyAction.StealingTech) // There are new techs to steal!
```

Comment:
Note that we don't actually reset progressTowardsStealingTech when beginning stealing technology again.